w3lib/url.py

```diff
@@ -199,13 +199,23 @@ def url_query_cleaner(url, parameterlist=(), sep='&', kvsep='=', remove=False, u
         url += '#' + fragment
     return url
 
-
 def _add_or_replace_parameters(url, params):
     parsed = urlsplit(url)
-    args = parse_qsl(parsed.query, keep_blank_values=True)
-
-    new_args = OrderedDict(args)
-    new_args.update(params)
+    current_args = parse_qsl(parsed.query, keep_blank_values=True)
+    new_args = []
+    changing_params = set(params)
+    seen_params = set()
+
+    for name, value in current_args:
+        if name in params:
+            if name not in seen_params:
+                new_args.append((name, params[name]))
+            seen_params.add(name)
```

Comment:
:lipstick: It does not matter much, but…
```suggestion
                seen_params.add(name)
```

Reply:
that's good 💅 :)